packages/mysql/src/mysql-adapter.ts

```diff
@@ -511,7 +511,17 @@ export class MySQLDatabaseQueryFactory extends SQLDatabaseQueryFactory {
 }
 
 export class MySQLDatabaseAdapter extends SQLDatabaseAdapter {
-    protected pool = createPool({ multipleStatements: true, maxAllowedPacket: 16_000_000, ...this.options });
+    protected pool = createPool({
+        multipleStatements: true,
+        maxAllowedPacket: 16_000_000,
+
+        // https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/master/documentation/callback-api.md#migrating-from-2x-or-mysqlmysql2-to-3x
+        insertIdAsNumber: true,
+        decimalAsNumber: true,
+        bigIntAsNumber: true,
```

Comment:
I'm sure this interferes with current mysql deserialiser which works fine with BigInts. I wonder why not tests break though. We probably don't have a bigint test yet that shows that precision is maintained.

Reply:
So you're right in that BigInt support is broken when this is enabled, but it would appear it's already broken -- which makes sense since the MariaDB docs say enabling those options restore the 2.x functionality.

Check out the [test I just added](https://github.com/deepkit/deepkit-framework/pull/526/commits/1256e31af97a8c57eb4864eba9d91a9d3e0ed25d#diff-54961ebf32e097acbde9d8dfb34f3f9cc2ac955824968a2eba9968c05f78acfdR215-R217), notably the comment about when it passes vs fails.

The issue we run into when letting BigInts remain as native BigInts (the default behavior of the 3.x driver) is that the MySQL `COUNT` function returns `BIGINT`, so suddenly every implementation of a counting function is using the wrong type.

Here's an example from the tests, without even considering how badly it may mess up people's existing application code:
```
  ● aggregate:basics

    expect(received).toEqual(expected) // deep equality

    - Expected  - 2
    + Received  + 2

      Array [
        Object {
    -     "amount": 1,
    +     "amount": 1n,
          "category": "planets",
        },
        Object {
    -     "amount": 2,
    +     "amount": 2n,
          "category": "toys",
        },
      ]

      25 |
      26 |         expect(await query.withSum('price').find()).toEqual([{ price: 999 + 499 + 45549 }]);
    > 27 |         expect(await query.groupBy('category').withCount('id', 'amount').orderBy('category').find()).toEqual([{ category: 'planets', amount: 1 }, { category: 'toys', amount: 2 }]);
         |                                                                                                      ^
      28 |         expect(await query.groupBy('category').withSum('price').orderBy('category').find()).toEqual([{ category: 'planets', price: 45549 }, {
      29 |             category: 'toys',
      30 |             price: 999 + 499

      at Object.basics (../orm-integration/src/aggregate.ts:27:102)
      at Object.<anonymous> (../orm-integration/index.ts:73:13)
```

My proposal would be to move forward with the driver upgrade with the options to maintain the 2.x functionality enabled, then we look at solutions for dealing with the type change for `COUNT` and any other functions that may be impacted (I would _assume_ the other aggregate functions maintain the type of the column they're aggregating).